tests/test_pandas.py

```diff
@@ -49,6 +52,29 @@ def test_types_to_datetime() -> None:
     )
 
 
+def test_types_concat_none() -> None:
+    """Test concatenation with None values."""
+    series = pd.Series([7, -5, 10])
+    df = pd.DataFrame({"a": [7, -5, 10]})
+
+    check(assert_type(pd.concat([None, series]), pd.Series), pd.Series)
+    check(assert_type(pd.concat([None, df]), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(pd.concat([None, series, df], axis=1), pd.DataFrame), pd.DataFrame
+    )
+
+    check(assert_type(pd.concat({"a": None, "b": series}), pd.Series), pd.Series)
+    check(assert_type(pd.concat({"a": None, "b": df}), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(pd.concat({"a": None, "b": series, "c": df}, axis=1), pd.DataFrame),
+        pd.DataFrame,
+    )
+
+    if TYPE_CHECKING_INVALID_USAGE:
+        assert_type(pd.concat({"a": None}), Never)
+        assert_type(pd.concat([None]), Never)
```

Comment:
I think these tests should be of the form:
```python
    pd.concat({"a": None})  # type: ignore[some_mypy_error] # pyright: ignore[some_pyright_error]
    pd.concat([None])  # type: ignore[some_mypy_error] # pyright: ignore[some_pyright_error]
```
so we are checking that the type checkers see that invalid code as an error.

Reply:
I believe I tried that first - the issue is that the second call will not be checked as it cannot be reached (by the type checkers). Would need to split it into two functions.

Reply:
That makes sense.  I think for consistency's sake, we should do that, although I could be convinced otherwise.  If we want to use your pattern here, then add a comment to indicate why we can't just check for a specific type checker error based on your comment here.

Reply:
I think we need the `assert_type` only when testing at least two invalid function calls where the first one "returns"  `NoReturn/Never`. I think we also have many cases were we do not explicitly return NoReturn/Never. In that case, we might not need the `assert_type`.

I would be inclined to use the `assert_type` only when we have to.